qiita_ware/ebi.py

```diff
@@ -885,8 +866,14 @@ def generate_demultiplexed_fastq(self, rewrite_fastq=False, mtime=None):
         """
         ppd = self.preprocessed_data
 
-        if not isdir(self.ebi_dir) or rewrite_fastq:
-            makedirs(self.ebi_dir)
+        dir_not_exists = not isdir(self.full_ebi_dir)
+        if dir_not_exists or rewrite_fastq:
+            if dir_not_exists:
+                makedirs(self.full_ebi_dir)
+                filetype_id = convert_to_id('directory', 'filepath_type')
+                insert_filepaths([(self.full_ebi_dir, filetype_id)], -1,
```

Comment:
This is just inserting the filepath in the `filepath` table, but it is not linking it with the preprocessed data.
I recommend to use PreprocessedData.add_filepath